src/components/Viewer.vue

```diff
@@ -0,0 +1,148 @@
+<script setup>
+/* eslint-disable */
+import { useAsyncState, useUrlSearchParams } from '@vueuse/core'
+import { Source } from 'rdf-cube-view-query'
+import { defineEmits, defineProps, onMounted, ref, watch, shallowRef} from 'vue'
+import { getBoundedViewPointer } from './common/debug.js'
+import { viewFromCubeUri, viewFromDataset, viewFromViewUri } from './common/viewLoaders.js'
+import LoadingIcon from './icons/LoadingIcon.vue'
+import MetadataHeader from './MetadataHeader.vue'
+import ShareUrlButton from './ShareUrlButton.vue'
+import TabularView from './TabularView.vue'
+
+const DELAY = 0
+
+const props = defineProps({
+  source: {
+    type: Source,
+    required: true
+  },
+  viewInput: {
+    type: Object
+  }
+})
+
+const emit = defineEmits(['setViewInput'])
+const view = shallowRef(null)
+
+onMounted(async () => {
+  await initFromProps()
+})
+
+async function fetchView ({
```

Comment:
This component loads a view, using cubeUri, viewUri or dataset and displays it using a TabularView